### ISSUE TITLE
Regression: explicitly qualified function calls in \usage{} break pkgdown

### ISSUE THREAD

**Issue by klmr** [r-lib/pkgdown#2750]
I have a package which contains the following `man/foo.Rd` file:

```r
\name{foo}
\alias{foo}
\title{A foo function}
\usage{
\special{xyz::foo()}
}
```

Until the last installed version of ‘pkgdown’ (2.0.7, IIRC), this wasn’t an issue. However, in version 2.1.0, running `pkgdown::build_reference()` results in the following error:

```
> pkgdown::build_reference()
── Building function reference ────────────────────────────────────────────────────────────
Reading man/foo.Rd
Error in `build_reference()`:
! Failed to parse Rd in foo.Rd
Caused by error in `is_call(x) && !is_call(x, "=") && !is_syntactic(x[[1]])`:
! 'length = 3' in coercion to 'logical(1)'
Run `rlang::last_trace()` to see where the error occurred.

> rlang::last_trace(drop = FALSE)
<error/rlang_error>
Error in `build_reference()`:
! Failed to parse Rd in foo.Rd
Caused by error in `is_call(x) && !is_call(x, "=") && !is_syntactic(x[[1]])`:
! 'length = 3' in coercion to 'logical(1)'
---
Backtrace:
     ▆
  1. ├─pkgdown::build_reference()
  2. │ ├─pkgdown:::unwrap_purrr_error(...)
  3. │ │ └─base::withCallingHandlers(...)
  4. │ └─purrr::map(...)
  5. │   └─purrr:::map_("list", .x, .f, ..., .progress = .progress)
  6. │     ├─purrr:::with_indexed_errors(...)
  7. │     │ └─base::withCallingHandlers(...)
  8. │     ├─purrr:::call_with_cleanup(...)
  9. │     └─pkgdown (local) .f(.x[[i]], ...)
 10. │       ├─base::withCallingHandlers(...)
 11. │       └─pkgdown:::data_reference_topic(...)
 12. │         ├─pkgdown:::as_data(tags$tag_usage[[1]])
 13. │         └─pkgdown:::as_data.tag_usage(tags$tag_usage[[1]])
 14. │           └─base::vapply(parsed, needs_tweak, logical(1))
 15. │             └─pkgdown (local) FUN(X[[i]], ...)
 16. └─base::.handleSimpleError(...)
 17.   └─pkgdown (local) h(simpleError(msg, call))
 18.     └─cli::cli_abort(...)
 19.       └─rlang::abort(...)
```

<details>
<summary>Output of <code>sessionInfo()</code></summary>

```
R version 4.3.3 (2024-02-29)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Monterey 12.7.5

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_GB.UTF-8/en_GB.UTF-8/en_GB.UTF-8/C/en_GB.UTF-8/en_GB.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base

other attached packages:
[1] xyz_0.0.1.9000

loaded via a namespace (and not attached):
 [1] crayon_1.5.3      vctrs_0.6.5       cli_3.6.3         knitr_1.48
 [5] rlang_1.1.4       xfun_0.46         pkgload_1.4.0     purrr_1.0.2
 [9] glue_1.7.0        rprojroot_2.0.4   htmltools_0.5.8.1 pkgbuild_1.4.4
[13] fansi_1.0.6       rmarkdown_2.27    evaluate_0.24.0   tibble_3.2.1
[17] fastmap_1.2.0     lifecycle_1.0.4   memoise_2.0.1     whisker_0.4.1
[21] compiler_4.3.3    fs_1.6.4          downlit_0.4.4     pkgconfig_2.0.3
[25] rstudioapi_0.16.0 digest_0.6.36     R6_2.5.1          utf8_1.2.4
[29] pillar_1.9.0      magrittr_2.0.3    tools_4.3.3       withr_3.0.1
[33] pkgdown_2.1.0     xml2_1.3.6        cachem_1.1.0      desc_1.4.3
```

</details>

<details>
<summary>Complete MCVE</summary>

```r
usethis::create_package('xyz', open = TRUE)
foo_rd = c('\\name{foo}', '\\alias{foo}', '\\title{A foo function}', '\\usage{', '\\special{xyz::foo()}', '}')
dir.create('man')
writeLines(foo_rd, 'man/foo.Rd')

pkgdown::build_reference()
```

</details>

**Comment by hadley** [comment]
Have you tried dev pkgdown? I think this was fixed in #2728

**Comment by klmr** [comment]
Ah! Brilliant, thanks, it works. I had obviously forgotten to test against dev — and I had searched for matching open issues but neglected closed ones.
